### PR TITLE
[JENKINS-43887] Update parent POM to 2.X.

### PR DESCRIPTION
**Pull request by varyvol**
[JENKINS-43887](https://issues.jenkins-ci.org/browse/JENKINS-43887)

- Uses 2.26 parent POM.
- Bump some dependencies.
- Fix tests.
- Fix Findbugs errors.

@reviewbybees @jglick @kwhetstone @andresrc @imod @bap2000

### CODE REVIEW

**Comment by kwhetstone** [comment]
:bee:

File: src/main/java/org/jenkinsci/plugins/conditionalbuildstep/ConditionalBuildStepHelper.java [hunk 1]
```diff
@@ -111,7 +111,12 @@ private ConditionalBuildStepHelper() {
      * @return
      */
     public static boolean isMavenPluginInstalled() {
-        final hudson.Plugin plugin = Jenkins.getInstance().getPlugin("maven-plugin");
-        return plugin != null ? plugin.getWrapper().isActive() : false;
+        Jenkins j = Jenkins.getInstance();
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/conditionalbuildstep/ConditionalBuildStepHelper.java]: Can just use `getActiveInstance`.

Comment (varyvol) [inline src/main/java/org/jenkinsci/plugins/conditionalbuildstep/ConditionalBuildStepHelper.java]: It's not available, maybe it was introduced after 1.580.1?

Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/conditionalbuildstep/ConditionalBuildStepHelper.java]: IIRC the first release with this method is 1.596.

**Comment by varyvol** [comment]
@imod could you review and merge this if appropriate?

**Comment by imod** [comment]
@varyvol thanks!

**Comment by varyvol** [comment]
@imod would it be possible to release a new version including these changes?

**Comment by imod** [comment]
@varyvol done...

**Comment by varyvol** [comment]
@imod thank you very much!

**Pull request merged**